### PR TITLE
Add `kueue.x-k8s.io/pod-group-fast-admission` annotation on docs.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Add `kueue.x-k8s.io/pod-group-fast-admission` annotation on docs.

#### Which issue(s) this PR fixes:

Part of #2717

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
Actually, I offered @vladikkuzn to add documentation here: https://github.com/kubernetes-sigs/kueue/pull/3189#issuecomment-2402101457

If @vladikkuzn is ok, we can move this forward, but, in that case, could you add a documentation to https://kueue.sigs.k8s.io/docs/tasks/run/plain_pods/#running-a-group-of-pods-to-be-admitted-together as well?

File: site/content/en/docs/reference/labels-and-annotations.md [hunk 1]
```diff
@@ -73,6 +73,17 @@ Used on: [MultiKueue](/docs/concepts/multikueue/).
...
 
+### kueue.x-k8s.io/pod-group-fast-admission
+
+Type: Annotation
+
+Example: `kueue.x-k8s.io/pod-group-fast-admission: "true"`
+
+Used on: [Plain Pods](/docs/tasks/run/plain_pods/).
+
+The annotation key is used to admit the plain pod, even if all the PodGroup pods haven't been created yet.
```
Comment (mimowo) [inline site/content/en/docs/reference/labels-and-annotations.md:84]:
```suggestion
The annotation key is used to allow admitting a PodGroup as soon as the first pod in the group is created.
```

**Comment by mimowo** [comment]
> Actually, I offered @vladikkuzn to add documentation here: https://github.com/kubernetes-sigs/kueue/pull/3189#issuecomment-2402101457

I agree, we should mention it there too. @vladikkuzn please either update this PR, or send another one.

File: site/content/en/docs/reference/labels-and-annotations.md (hunk 1 above)
Comment (mbobrovskyi) [reply site/content/en/docs/reference/labels-and-annotations.md]: Done

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
I think the PR is ready to go, let's update the Pods documentation in a follow up @vladikkuzn

**Pull request merged**